blockchain/state/iterator.go

```diff
@@ -186,15 +185,152 @@ func (it *NodeIterator) retrieve() bool {
 	return true
 }
 
+// CheckStateConsistencyParallel checks parallelly the consistency of all state/storage trie of given two state database.
+func CheckStateConsistencyParallel(oldDB Database, newDB Database, root common.Hash, quitCh chan struct{}) error {
+	// Create and iterate a state trie rooted in a sub-node
+	oldState, err := New(root, oldDB)
+	if err != nil {
+		return err
+	}
+
+	_, err = New(root, newDB)
+	if err != nil {
+		return err
+	}
+
+	children, err := oldState.db.TrieDB().NodeChildren(root)
```

Comment:
Why can't it be just like this?
```suggestion
	children, err := oldDB.TrieDB().NodeChildren(root)
```

Reply:
reflected